screenwriter/screenwriter.go

```diff
@@ -0,0 +1,60 @@
+package screenwriter
+
+import (
+	"fmt"
+	"io"
+	"strings"
+)
+
+type ScreenWriter struct {
+	writer   io.StringWriter
+	previous string
+}
+
+func New(writer io.StringWriter) *ScreenWriter {
+	return &ScreenWriter{
+		writer: writer,
+	}
+}
+
+func (w *ScreenWriter) ClearPrevious() error {
+	if w.previous == "" {
+		return nil
+	}
+
+	// Get number of lines in previous
+	lines := strings.Count(w.previous, "\n")
+
+	// Handle the final line
+	finalLineLength := 0
+	splitLines := strings.Split(w.previous, "\n")
+	finalLine := splitLines[len(splitLines)-1]
+	finalLineLength = len(finalLine)
+	w.writer.WriteString(fmt.Sprintf("\u001b[%dD", finalLineLength))
+
+	// Clear entire line
+	// This is necessary if the last line has content (i.e. the string does not end with \n)
+	w.writer.WriteString("\u001b[2K")
```

Comment:
It might be nicer to declare these character codes as constants with descriptive names, so they're less magical, and are less likely to lose their commented labels through refactoring.

Reply:
That would definitely help -- I only know what these do b/c I've previously written abstractions for it so I can look up what each code does.

Reply:
I've addressed this in my most recent commit.